compass/landice/tests/greenland/decomposition_test/__init__.py

```diff
@@ -10,22 +10,33 @@ class DecompositionTest(TestCase):
     results of the two runs are identical.
     """
 
-    def __init__(self, test_group):
+    def __init__(self, test_group, velo_solver):
         """
         Create the test case
 
         Parameters
         ----------
         test_group : compass.landice.tests.greenland.Greenland
             The test group that this test case belongs to
+
+        velo_solver : {'sia', 'FO'}
+            The velocity solver to use for the test case
         """
         name = 'decomposition_test'
-        super().__init__(test_group=test_group, name=name)
+        self.velo_solver = velo_solver
+        subdir = '{}_{}'.format(velo_solver.lower(), name)
+        super().__init__(test_group=test_group, name=name, subdir=subdir)
+
+        if velo_solver == 'sia':
+            self.cores_set = [1, 8]
+        elif velo_solver == 'FO':
+            self.cores_set = [16, 32]
```

Comment:
```suggestion
        elif velo_solver == 'FO':
            self.cores_set = [16, 32]
        else:
            raise ValueError('Unexpected velo_solver {}'.format(velo_solver))
```
It's best to fail here rather than in an ambiguous way later on (e.g. a variable that isn't defined).

Reply:
The same for the other 2 test cases.

Reply:
Good suggestion.  I'm not up to speed on what error to raise, so this suggestion was perfect.